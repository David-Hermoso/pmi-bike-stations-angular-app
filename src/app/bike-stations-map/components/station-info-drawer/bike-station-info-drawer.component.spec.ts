import {BikeStationInfoDrawerComponent} from './bike-station-info-drawer.component';

describe('StationInfoDrawerComponent', () => {
  const deviceDetectorMock = {
    isDesktop: () => false
  };
  const component: BikeStationInfoDrawerComponent = new BikeStationInfoDrawerComponent(deviceDetectorMock as any);


  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('isDesktopDevice should be true', () => {
    expect(component).toBeTruthy();
  });

  it('getDirectionsHref: should return url', () => {
    component.bikeStation = {
      img: null,
      id: '77',
      name: 'SON COSTA - SON FORTEZA',
      lng: 2.6661,
      lat: 39.584114,
      fullAddress: 'Argelaga, 26, 07005 Palma, Illes Balears',
      street: 'Argelaga',
      streetNumber: 26,
      cp: '07005',
      town: 'Palma',
      region: 'Illes Balears'
    };

    expect(component.getDirectionsHref()).toMatchSnapshot();
  });

  it('getDirectionsHref: should return falsy value', () => {
    component.bikeStation = undefined;

    expect(component.getDirectionsHref()).toBeFalsy();
  });
});
