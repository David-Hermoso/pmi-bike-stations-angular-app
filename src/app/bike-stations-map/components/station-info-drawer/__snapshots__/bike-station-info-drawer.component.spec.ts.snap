// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationInfoDrawerComponent getDirectionsHref: should return url 1`] = `"https://www.google.es/maps/dir/39.598099,2.6788515/39.584114,2.6661/data=!3m1!4b1!4m2!4m1!3e1"`;
